service/handlers.go

```diff
@@ -981,6 +981,81 @@ func (h handler) updateTarget(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprint(w, string(data))
 }
 
+// Creates a token
+func (h handler) createToken(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	projectName := vars["projectName"]
+
+	l := h.requestLogger(r, "op", "create-token", "project", projectName)
+
+	level.Debug(l).Log("message", "validating authorization header for token list")
+	ah := r.Header.Get("Authorization")
+	a, err := credentials.NewAuthorization(ah)
+	if err != nil {
+		h.errorResponse(w, "error unauthorized, invalid authorization header format", http.StatusUnauthorized)
+		return
+	}
+	if err := a.Validate(a.ValidateAuthorizedAdmin(h.env.AdminSecret)); err != nil {
+		h.errorResponse(w, "error unauthorized, invalid authorization header", http.StatusUnauthorized)
+		return
+	}
+
+	ctx := r.Context()
+
+	level.Debug(l).Log("message", "creating credential provider")
+	cp, err := h.newCredentialsProvider(*a, h.env, r.Header, credentials.NewVaultConfig, credentials.NewVaultSvc)
+	if err != nil {
+		level.Error(l).Log("message", "error creating credentials provider", "error", err)
+		h.errorResponse(w, "error creating credentials provider", http.StatusInternalServerError)
+		return
+	}
+	level.Debug(l).Log("message", projectName)
+	projectExists, err := cp.ProjectExists(projectName)
```

Comment:
@Acepie I know in one of your PR's, we decided to look at the DB to check if the project exists. Should we do that here too? I could go either way since the creds provider is used later on. We're kinda in a weird spot in terms of what is the "source of truth". Leaning towards having the DB be the source of truth in case the creds provider no longer stores data (only returns data) in the future.

Reply:
Yeah i think using the db makes sense (especially since i know long term we theoretically wanna support different cred providers). Also if we are going to do that then i think we should make the db interface have a dedicated projectExists method

Reply:
I don't think we can switch to exclusively a DB check at this time. We still need to check the credential provider is properly configured before attempting to create a token.